http/main.go

```diff
@@ -10,116 +10,148 @@ import (
 	item := r.FormValue("item")
-	if item == "" {
+	switch item {
+	case "":
 		http.Error(w, "No item given", http.StatusBadRequest)
 		return
+	default:
+		p.Lock()
+		p.db = append(p.db, item)
+		p.Unlock()
 	}
 
-	priceStr := r.FormValue("price")
-	price, err := strconv.Atoi(priceStr)
-	if err != nil {
-		http.Error(w, "No integer price given", http.StatusBadRequest)
-		return
-	}
-
-	if _, ok := p.db[item]; ok {
-		http.Error(w, fmt.Sprintf("%s already exists", item), http.StatusBadRequest)
-		return
-	}
-
-	p.Lock()
-	if p.db == nil {
-		p.db = make(map[string]int, 0)
-	}
-	p.db[item] = price
-	p.Unlock()
 }
 
-func (p *PriceDB) Update(w http.ResponseWriter, r *http.Request) {
+func (p *DB) Update(w http.ResponseWriter, r *http.Request) {
+	///update?id=2&item=zz
+	idc := r.FormValue("id")
+	id, _ := strconv.Atoi(idc)
 	item := r.FormValue("item")
-	if item == "" {
-		http.Error(w, "No item given", http.StatusBadRequest)
+
+	if len(p.db)-1 < id {
+		http.Error(w, "no item of that id", http.StatusBadRequest)
 		return
 	}
 
-	priceStr := r.FormValue("price")
-	price, err := strconv.Atoi(priceStr)
-	if err != nil {
-		http.Error(w, "No integer price given", http.StatusBadRequest)
+	if idc == "" {
+		http.Error(w, "No id given", http.StatusBadRequest)
 		return
 	}
 
-	if _, ok := p.db[item]; !ok {
-		http.Error(w, fmt.Sprintf("%s doesn't exist", item), http.StatusNotFound)
+	switch item {
+	case "":
+		http.Error(w, "No item given", http.StatusBadRequest)
 		return
+	default:
+		p.Lock()
+		//update
+		for item, _ := range p.db {
+			p.db = append(p.db[:item], p.db[item:]...)
+
+		}
+		p.Unlock()
 	}
 
-	p.Lock()
-	p.db[item] = price
-	p.Unlock()
 }
 
-func (p *PriceDB) Delete(w http.ResponseWriter, r *http.Request) {
-	item := r.FormValue("item")
-	if item == "" {
-		http.Error(w, "No item given", http.StatusBadRequest)
+// 	item := r.FormValue("item")
+// 	if item == "" {
+// 		http.Error(w, "No item given", http.StatusBadRequest)
+// 		return
+// 	}
+
+// 	priceStr := r.FormValue("price")
+// 	price, err := strconv.Atoi(priceStr)
+// 	if err != nil {
+// 		http.Error(w, "No integer price given", http.StatusBadRequest)
+// 		return
+// 	}
+
+// 	if _, ok := p.db[item]; !ok {
+// 		http.Error(w, fmt.Sprintf("%s doesn't exist", item), http.StatusNotFound)
+// 		return
+// 	}
+
+// 	p.Lock()
+// 	p.db[item] = price
+// 	p.Unlock()
+// }
+
+func (p *DB) Delete(w http.ResponseWriter, r *http.Request) {
+
+	idc := r.FormValue("id")
+	id, _ := strconv.Atoi(idc)
+	if len(p.db)-1 < id {
+		http.Error(w, "no particular id", http.StatusBadRequest)
 		return
 	}
 
-	if _, ok := p.db[item]; !ok {
-		http.Error(w, fmt.Sprintf("%s doesn't exist", item), http.StatusNotFound)
+	switch idc {
+	case "":
+		http.Error(w, "No id given", http.StatusBadRequest)
 		return
-	}
 
-	p.Lock()
-	delete(p.db, item)
-	p.Unlock()
+	default:
+		p.Lock()
+		id, _ := strconv.Atoi(idc)
+		sl := p.db
+		sl = append(sl[0:id], sl[id+1:]...)
+		p.db = sl
+		p.Unlock()
```

Comment:
I think you could have just gone with `p.db = append(p.db[:id], p.db[id+1:]...)`
See if you can find a more efficient way to remove an element from a slice.

Reply:
I will try that, thanks